root/usr/lib64/ruby/vendor_ruby/discovery/menu.rb

```diff
@@ -20,32 +21,41 @@ def enable_root_account(pass)
 def error_box(msg, extra_msg = nil)
   log_err msg
   log_err extra_msg
-  backtrace = log_exception extra_msg
+  backtrace = log_exception(extra_msg).gsub(/^\/.*vendor_ruby\//, '')
   Newt::Screen.centered_window(74, 20, "Fatal error")
   f = Newt::Form.new
   t_desc = Newt::Textbox.new(1, 1, 70, 13, Newt::FLAG_SCROLL)
   random_pass = (1...10).map { (65 + rand(26)).chr }.join
-  t_desc.set_text "#{msg}:\n#{extra_msg}\n\n#{backtrace}\n\n" +
+  t_desc.set_text "#{msg}:\n#{extra_msg}\n\n" +
   "Once OK button is pressed, and root account will be unlocked\n" +
   "with the following random password:\n\n" +
   "   #{random_pass}\n\n" +
   "It is possible to use third console to login and investigate\n" +
-  "via journalctl and discovery-debug commands. "
+  "via journalctl and discovery-debug commands.\n\n" +
+  "PIPELINE:\n#{Pipeline.instance}\n\nBACKTRACE:\n#{backtrace}"
   b_ok = Newt::Button.new(66, 15, "OK")
   f.add(b_ok, t_desc)
   f.run
   enable_root_account(random_pass)
   exit(1)
 end
 
-def command(cmd, fail_on_error = true, send_to_syslog = true)
-  log_msg("TUI executing: #{cmd}") if send_to_syslog
-  # do not run real commands in development (non-image) environment
-  return true if fdi_version == 'GIT'
-  output = `#{cmd} 2>&1`
+def command(cmd, fail_on_error = true, cmd_syslog = true, output_syslog = false, redirect_stdout = true)
+  cmd = "systemd-cat #{cmd}" if output_syslog
+  cmd = "#{cmd} 2>&1" if redirect_stdout
+  if ENV['FDI_DEVEL']
+    # do not run real commands in development (non-image) environment
+    log_msg("*FAKE*: #{cmd}")
+    output = "Fake output (commands disabled)"
+    sleep 0.1
+    return true
+  else
+    log_msg("*EXEC*: #{cmd}") if cmd_syslog
+  end
```

Comment:
This help development of TUI, you can simply start it from git locally and if FDI_DEVEL is set, all commands executed are actually just printed into stdout.